x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/RestVectorTileAction.java

```diff
@@ -274,6 +297,21 @@ private static VectorTile.Tile.Layer.Builder buildHitsLayer(SearchHit[] hits, Ve
                 }
                 hitsLayerBuilder.addFeatures(featureBuilder);
             }
+            if (request.getWithLabels()) {
+                final DocumentField labelField = searchHit.field(LABEL_POSITION_FIELD_NAME);
+                if (labelField != null) {
+                    Object labelPosValue = labelField.getValue();
+                    GeoPoint labelPos = GeoUtils.parseGeoPoint(labelPosValue, true);
+                    byte[] labelPosFeature = featureFactory.point(labelPos.lon(), labelPos.lat());
+                    if (labelPosFeature != null && labelPosFeature.length != 0) {
+                        featureBuilder.clear();
+                        featureBuilder.mergeFrom(labelPosFeature);
+                        VectorTileUtils.addPropertyToFeature(featureBuilder, layerProps, ID_TAG, searchHit.getId());
+                        VectorTileUtils.addPropertyToFeature(featureBuilder, layerProps, LABEL_POSITION_TAG, true);
+                        hitsLayerBuilder.addFeatures(featureBuilder);
```

Comment:
I think we agreed that we should add all the tags that we added to the feature that is not a tag. That brings the question to what happens if the user has defined an attribute with the same name as the label?

Reply:
I think both concerns are considered in https://github.com/elastic/elasticsearch/pull/86458/commits/9fa8c0028b756e904beeeddc8a95c0cc4707fbe4